src/CodeGenerator/Generator/TestGenerator.php

```diff
@@ -0,0 +1,324 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\CodeGenerator\Generator;
+
+use AsyncAws\CodeGenerator\Definition\ListShape;
+use AsyncAws\CodeGenerator\Definition\MapShape;
+use AsyncAws\CodeGenerator\Definition\Operation;
+use AsyncAws\CodeGenerator\Definition\Shape;
+use AsyncAws\CodeGenerator\Definition\StructureMember;
+use AsyncAws\CodeGenerator\Definition\StructureShape;
+use AsyncAws\CodeGenerator\File\FileWriter;
+use AsyncAws\CodeGenerator\Generator\Naming\ClassName;
+use AsyncAws\CodeGenerator\Generator\Naming\NamespaceRegistry;
+use AsyncAws\CodeGenerator\Generator\PhpGenerator\ClassFactory;
+use AsyncAws\Core\Test\Http\SimpleMockedResponse;
+use AsyncAws\Sqs\Result\CreateQueueResult;
+use Nette\PhpGenerator\ClassType;
+use Nette\PhpGenerator\PhpNamespace;
+use Symfony\Component\HttpClient\MockHttpClient;
+
+/**
+ * Generate API test methods.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ *
+ * @internal
+ */
+class TestGenerator
+{
+    /**
+     * @var NamespaceRegistry
+     */
+    private $namespaceRegistry;
+
+    /**
+     * @var FileWriter
+     */
+    private $fileWriter;
+
+    public function __construct(NamespaceRegistry $namespaceRegistry, FileWriter $fileWriter)
+    {
+        $this->namespaceRegistry = $namespaceRegistry;
+        $this->fileWriter = $fileWriter;
+    }
+
+    /**
+     * Update the API test client
+     */
+    public function generate(Operation $operation): void
+    {
+        $this->generateIntegration($operation);
+        $this->generateInput($operation, $operation->getInput());
+        if (null !== $result = $operation->getOutput()) {
+            $this->generateResult($operation, $result);
+        }
+    }
+
+    private function generateInput(Operation $operation, StructureShape $shape)
+    {
+        $className = $this->namespaceRegistry->getInputUnitTest($shape);
+        $methodName = 'testRequestBody';
+        $flagSkipped = 'self::markTestSkipped(\'Not implemented\');';
+        try {
+            $namespace = ClassFactory::fromExistingClass($className->getFqdn());
+            $classes = $namespace->getClasses();
+            $class = $classes[\array_key_first($classes)];
+
+            if ($class->hasMethod($methodName) && \strpos($class->getMethod($methodName)->getBody(), $flagSkipped) === false) {
+                return;
```

Comment:
So dont generate the test if method exists and method does not have "Not Implmented"

:+1:

Reply:
maybe `hasMethod` is enougth.

Reply:
No. I like this.

Reply:
removed. 
Issue is: When loading an existing class, Nette broke the indentation of some part of code.